### ISSUE TITLE
Mouse follows focus off, still switches screen focus

### ISSUE THREAD

**Issue by milch** [koekeishiya/kwm#364]
Hi,

I have turned off focus follows mouse using the following commands:

```
kwmc config mouse-follows-focus off
kwmc config focus-follows-mouse off
```

This works, and when I hover over a window it doesn't gain focus anymore. However, when I move my  mouse between screens, it seems that kwm clicks the left mouse button once (e.g. if the mouse hovers over to the menu bar, the menu drops down), so that screen gains focus. Is there any way to turn this behaviour off?

**Comment by koekeishiya** [comment]
> it seems that kwm clicks the left mouse button once (e.g. if the mouse hovers over to the menu bar, the menu drops down), so that screen gains focus. 

This is indeed what happens. It will only happen one time per space of that monitor.
This can not be turned off as it will break window tiling with multiple monitors.

**Comment by milch** [comment]
Hm okay, thanks for getting back to me! 

My problem with this is just that an email was sent by mistake because of this, since the click triggered when it was directly above the 'send' button in my mail program. In this case the mail was already finished and it was no problem, but since this is the first day I'm using kwm it isn't hard to imagine that this could potentially happen again, and then the mail could be in a more unfinished state. 

Maybe something could be done so that a scenario like the above doesn't happen? For example, is the click really needed, or does the screen just need to gain focus? Otherwise, could the click be delayed by some heuristic, e.g. could it be detected that there is a button under the screen coordinates where the click will be triggered? Could the click always be triggered at some screen coordinates where a button is unlikely to be, or does it have to come from the actual mouse pointer?

**Comment by koekeishiya** [comment]
> My problem with this is just that an email was sent by mistake because of this, since the click triggered when it was directly above the 'send' button in my mail program. In this case the mail was already finished and it was no problem, but since this is the first day I'm using kwm it isn't hard to imagine that this could potentially happen again, and then the mail could be in a more unfinished state. 

That does indeed sound like a bad scenario.

> For example, is the click really needed, or does the screen just need to gain focus?

The issue really is that for the screen to gain focus (to report as the focused on from OSX APIs, which Kwm relies on), a mouse click has to be performed either by the user, or automatically by Kwm.

> Otherwise, could the click be delayed by some heuristic

Delaying the click wold unfortunately still mess up the tiling.

> could it be detected that there is a button under the screen coordinates where the click will be triggered?

This is also not possible to my knowledge.

> Could the click always be triggered at some screen coordinates where a button is unlikely to be, or does it have to come from the actual mouse pointer? 

I can indeed set the position of the mouse pointer before triggering a mouse click event.

**Comment by milch** [comment]
> I can indeed set the position of the mouse pointer before triggering a mouse click event.

That sounds good, can we try that? 

I don't know if this ties in to this issue or if I should create a new one for this, but I noticed that when I connect my third monitor, it seems that the click is *not* issued - and as you said, tiling is broken on that screen. When I use one of my shortcuts to move a window to that screen, it is instantly pulled back to the screen it was on before. I tried both with the latest release and the master branch (as of yesterday) and the issue is present in both.
